Web apps support

csdodd opened the pull request:
Add a new activity that is just a webview. 
Allow entering a URL in the search bar to trigger the webview. 
Inject the correct interface and JS to allow web apps to communicate with the client.

martintoften commented on app/src/main/res/layout/activity_web_view.xml:
```diff
@@ -0,0 +1,50 @@
+  -->
+
+<!--
+  ~ 	Copyright (c) 2017. Token Browser, Inc
```
You added 2 licenses here

martintoften commented on app/src/main/java/com/tokenbrowser/presenter/webview/SofaWebViewClient.java:
```diff
@@ -0,0 +1,69 @@
+        }
+    }
+
+    private String loadScriptFromAssets() throws IOException {
```
Should this run on a worker thread?

csdodd replied:
Yes. It's only a small file now but it may grow. Good catch. I'll fix this.

csdodd commented:
Updated. I added a toolbar at the same time. There's a bug in app search that is exposed by opening and closing a web app but I'll fix that in a seperate PR.

martintoften commented on app/src/main/java/com/tokenbrowser/presenter/webview/SofaWebViewClient.java:
```diff
@@ -18,52 +18,103 @@
+    private Single<String> getSofaScript() {
+        return
+                Single.fromCallable(() -> {
+                    while (sofaScript == null) {
```
Comment: Could this be replaced by this:
´´´
 Observable
            .interval(100, TimeUnit.MILLISECONDS)
            .filter(__ -> sofaScript != null)
            .map(__ -> return sofaScript)
            ...
´´´

martintoften commented on app/src/main/java/com/tokenbrowser/presenter/webview/SofaWebViewClient.java:
```diff
@@ -18,52 +18,103 @@
+    }
+
+    private void asyncLoadSofaScript() {
+        loadSofaScript()
```
Add the subscription to a ´CompositeSubscription`

martintoften commented on app/src/main/java/com/tokenbrowser/presenter/webview/SofaWebViewClient.java:
```diff
@@ -18,52 +18,103 @@
 
     @Override
     public void onPageFinished(final WebView webView, final String url) {
+        getSofaScript()
```
Add the subscription to a CompositeSubscription

csdodd commented:
Updated to clear subscriptions in `SofaWebView`

csdodd commented:
I'm not sure that `Observable.interval` is the best match for what I am trying to achieve. I could make it work but I'm not sure if the code would be clearer.
